### ISSUE TITLE
How to define a custom docker host url?

### ISSUE THREAD

**Issue by xetra11** [spotify/dockerfile-maven#112]
I'm missing the way to configure a remote docker host daemon to deploy to.
The other maven plugin provided a way to configure it. Now I cannot find a way to do so:

**Old Plugin:**
```
<configuration>
    <!-- Docker Host, Imagename und Dockerfile Directory -->
    <dockerHost>http://localhost:2375</dockerHost>
    <imageName>${project.build.finalName}</imageName>
    <dockerDirectory>.</dockerDirectory>
</configuration>
```

Am trying to use <repository> for this now but surely it's wrong. The README.md did not provide any informationen how to do it.

Can anybody point me out to the solution?

Charlie

**Comment by mattnworb** [comment]
The plugin uses `DefaultDockerClient.fromEnv()` which respects the DOCKER_HOST environment variable. See https://github.com/spotify/docker-client/blob/master/docs/user_manual.md#creating-a-docker-client

**Comment by paul-hammant** [comment]
"This plugin uses DefaultDockerClient.fromEnv() which respects the DOCKER_HOST environment variable."  ... would be a fantastic addition to the README.

**Comment by mattnworb** [comment]
@paul-hammant good idea. Want to make a PR to add it?

**Comment by dennis0721** [comment]
"DOCKER_HOST" does not work for me

my config just like:
![default](https://user-images.githubusercontent.com/23226155/42331582-57f4f564-80a8-11e8-884d-95fec845d639.png)
![default](https://user-images.githubusercontent.com/23226155/42331672-9b7294fe-80a8-11e8-979e-af8eae99a365.png)

exception:
![default](https://user-images.githubusercontent.com/23226155/42331785-e196bc76-80a8-11e8-8de4-0335a484d5f1.png)

why it connect to "localhost"

**Comment by j-bro** [comment]
I'm having this issue as well.

I set DOCKER_HOST to unix:///var/run/docker.sock, but the plugin still attempts to use unix://localhost:80 as the route to the docker daemon.
```
...
+ printenv | grep DOCKER
18:04:06 DOCKER_HOST=unix:///var/run/docker.sock
...
18:04:07 + mvn clean install
...
18:06:21 [INFO] Building Docker context /data/jenkins/workspace/webapp-docker
18:06:21 [INFO] 
18:06:21 [INFO] Image will be built as webapp-api:local-build
18:06:21 [INFO] 
18:06:26 Jul 12, 2018 4:06:25 PM com.spotify.docker.client.shaded.org.apache.http.impl.execchain.RetryExec execute
18:06:26 INFO: I/O exception (java.io.IOException) caught when processing request to {}->unix://localhost:80: Permission denied
18:06:26 Jul 12, 2018 4:06:25 PM com.spotify.docker.client.shaded.org.apache.http.impl.execchain.RetryExec execute
18:06:26 INFO: Retrying request to {}->unix://localhost:80
18:06:26 Jul 12, 2018 4:06:25 PM com.spotify.docker.client.shaded.org.apache.http.impl.execchain.RetryExec execute
18:06:26 INFO: I/O exception (java.io.IOException) caught when processing request to {}->unix://localhost:80: Permission denied
18:06:26 Jul 12, 2018 4:06:25 PM com.spotify.docker.client.shaded.org.apache.http.impl.execchain.RetryExec execute
18:06:26 INFO: Retrying request to {}->unix://localhost:80
18:06:26 Jul 12, 2018 4:06:25 PM com.spotify.docker.client.shaded.org.apache.http.impl.execchain.RetryExec execute
18:06:26 INFO: I/O exception (java.io.IOException) caught when processing request to {}->unix://localhost:80: Permission denied
18:06:26 Jul 12, 2018 4:06:25 PM com.spotify.docker.client.shaded.org.apache.http.impl.execchain.RetryExec execute
18:06:26 INFO: Retrying request to {}->unix://localhost:80
18:06:26 [WARNING] An attempt failed, will retry 1 more times
18:06:26 org.apache.maven.plugin.MojoExecutionException: Could not build image
18:06:26     at com.spotify.plugin.dockerfile.BuildMojo.buildImage (BuildMojo.java:185)
18:06:26     at com.spotify.plugin.dockerfile.BuildMojo.execute (BuildMojo.java:105)
18:06:26     at com.spotify.plugin.dockerfile.AbstractDockerMojo.tryExecute (AbstractDockerMojo.java:252)
18:06:26     at com.spotify.plugin.dockerfile.AbstractDockerMojo.execute (AbstractDockerMojo.java:241)
18:06:26     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
18:06:26     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
18:06:26     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
18:06:26     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
18:06:26     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
18:06:26     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
18:06:26     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
18:06:26     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
18:06:26     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
18:06:26     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
18:06:26     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
18:06:26     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
18:06:26     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
18:06:26     at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
18:06:26     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
18:06:26     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
18:06:26     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
18:06:26     at java.lang.reflect.Method.invoke (Method.java:498)
18:06:26     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
18:06:26     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
18:06:26     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
18:06:26     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
```

From the discussion in #123 it seems to be related to casing, but everything in my pom.xml file is lowercase so I haven't been able to use this resolution.

**Comment by mattnworb** [comment]
@j-bro the `unix://localhost:80` thing is a red herring as I believe the plugin is replacing `/var/run/docker.sock` _when displaying the DOCKER_HOST value_ for internal reasons.

Looks like your exception stacktrace is incomplete - can you post the whole thing?

What happens if you run `DOCKER_HOST=unix:///var/run/docker.sock docker ps`?

**Comment by j-bro** [comment]
@mattnworb You're right, and I figured it out from your comment [here](https://github.com/spotify/docker-client/issues/479#issuecomment-235606168).

My issue was actually the permissions on the socket, as evidenced by the "Permission denied" message in the stack trace. I failed to mention that I was running maven in a container and mounting the Docker socket as a volume. Once I added `--group-add 999` to the `docker run` args it worked great.

Thanks again!
